app/models/form.rb

```diff
@@ -175,9 +175,10 @@ def visible_questionings
     questionings.reject{|q| q.hidden?}
   end
 
-  # returns questionings that work with sms forms and are not hidden
+  # returns hash of questionings that work with sms forms and are not hidden
   def smsable_questionings
-    questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings = questionings.reject{|q| q.hidden? || !q.question.qtype.smsable?}
+    smsable_questionings.index_by.with_index { |_, i| i + 1 }
```

Comment:
The part that I'm having a hard time grokking is how the block given to `with_index` ends up being the block used by the `yield` in `index_by`. I've looked through both the relevant Ruby code and the C code and I still don't get it, though that's clearly my problem since it works great :confounded: 

I definitely think the `smsable_questionings.map.with_index { |q, i| [i + 1, q] }.to_h` code is clearer.